spec/operators/timeout-spec.ts

```diff
@@ -139,4 +139,27 @@ describe('Observable.prototype.timeout', () => {
     expectObservable(result).toBe(expected, values, value);
     expectSubscriptions(e1.subscriptions).toBe(e1subs);
   });
+
+  it('should unsubscribe from the scheduled timeout action when timeout is unsubscribed early', () => {
+    const e1 =   hot('--a--b--c---d--e--|');
+    const e1subs =   '^         !        ';
+    const expected = '--a--b--c--        ';
+    const unsub =    '          !        ';
+
+    const result = e1
+      .lift(function(source) {
```

Comment:
@trxcllnt ugh.. this is a case where we're abusing the fact that `Operator` has a `call` method, and we're therefor passing a function to `lift` here? This is really convoluted, it took me a while to figure that out. If anything this test reminded me why `Operator` probably shouldn't have a `call` method.

I think we need to find a different way to test this.